### PR TITLE
Restrict access to old routes

### PR DESCRIPTION
**Pull request by agolden**
This ensures that old routes (for days older than yesterday) will not be served up.

Also, updates test data with the most recent data upload schema.

### CODE REVIEW

File: pages/api/cck/route.gpx.ts [hunk 1]
```diff
@@ -18,7 +18,12 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
 
     mode = mode ? mode : "bicycling";
 
-    if(passcode) {
+    const daysDiff = daysSince(parseDashedDate(date as string));
```
Comment (pdl) [inline pages/api/cck/route.gpx.ts:21]: If `date` is not a valid 'dashed date', we should not continue. Currently, if this is the case, `daysDiff` is `NaN`, which is not greater than 1.

```
    const parsedDate = parseDashedDate(date as string)

    ...

    if(daysDiff > 1 || Number.isNaN(Number(parsedDate))) {
```

File: components/api-helpers/APIHelpers.ts [hunk 2]
```diff
@@ -12,6 +12,25 @@ export enum UserRole {
     EVENT_ADMIN = 'event-admin'
 }
 
+/**
+ * Calculates the days since a provided date
```
Comment (pdl) [inline components/api-helpers/APIHelpers.ts:16]: I would add to this documentation:

> This returns a positive number for dates more than ~24h in the past and a negative number for dates more than ~24h in the future.

File: pages/api/cck/route.ts [hunk 3]
```diff
@@ -26,7 +27,12 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     const query = req.query;
     const { date, ref, passcode } = query;
 
-    if(passcode) {
+    const daysDiff = daysSince(parseDashedDate(date as string));
```
Comment (pdl) [inline pages/api/cck/route.ts:30]: Same comment as in the other file

File: components/api-helpers/APIHelpers.ts (hunk 2 above)
Comment (agolden) [reply components/api-helpers/APIHelpers.ts:16]: comments added

File: pages/api/cck/route.gpx.ts (hunk 1 above)
Comment (agolden) [reply pages/api/cck/route.gpx.ts]: Updated to throw error and return 400 if date cannot be successfully parsed

File: pages/api/cck/route.ts (hunk 3 above)
Comment (agolden) [reply pages/api/cck/route.ts]: Updated to throw error and return 400 if date cannot be successfully parsed

**Review by pdl — APPROVED** [review]
Thanks for the changes - looks good now.

**Pull request merged**